src/AWS/CRT/Auth/StaticCredentialsProvider.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace AWS\CRT\Auth;
+
+/**
+ * Provides a static set of AWS credentials
+ *
+ * @param array options:
+ * - string access_key_id - AWS Access Key Id
+ * - string secret_access_key - AWS Secret Access Key
+ * - string session_token - Optional STS session token
+ */
+final class StaticCredentialsProvider extends CredentialsProvider {
+
+    static function defaults() {
+        return array(
+            'access_key_id' => '',
+            'secret_access_key' => '',
+            'session_token' => '',
+        );
+    }
+
+    private $credentials = null;
+
+    public function __get($name) {
+        return $this->$name;
+    }
+
+    function __construct(array $options = array()) {
```

Comment:
similar to constructor above, our code convention is to use [] before array(), and remove the array as the param type

Reply:
Will fix. I'm amazed that habit came back to me from 20 years ago.